src/service/url-replacements-impl.js

```diff
@@ -459,16 +462,12 @@ export class GlobalVariableSource extends VariableSource {
     });
 
     // Returns the user agent client hint.
-    this.setAsync(
+    this.setBoth(
       'UACH',
-      (variable) =>
-        win.navigator?.userAgentData
-          ?.getHighEntropyValues([variable])
-          ?.then((values) =>
-            typeof values[variable] !== 'object'
-              ? values[variable]
-              : JSON.stringify(values[variable])
-          ) || Promise.resolve('')
+      // Synchronous alternative, will only work if the requested values were
+      // previously requested using the async method.
+      (variable) => this.cachedUach_[variable],
```

Comment:
is the "empty" return type consistent between async and sync? i see a `Promise.resolve('')` as a default return on the async side so i was wondering if the sync side will also always return empty string or can it be sometimes `undefined`? and if it is undefined are there side effect to that?